### PR TITLE
Document constants with comments, actual names, and more thorough usage

### PR DESCRIPTION
**Pull request by roukaour**
Constants are commented with the code and file they refer to.
Redundant constants like `NUM_WILDMONS_PER_AREA_TIME_OF_DAY` and `NUM_GRASSMON` have been merged.
sprite_constants.asm in particular has been filled out with meaningful names.

### CODE REVIEW

**Comment by Sanqui** [comment]
I love this

File: macros/charmap.asm [hunk 1]
```diff
@@ -152,7 +188,7 @@
 	charmap "♂",        $ef
 	charmap "¥",        $f0
 	charmap "×",        $f1
-	charmap "·",        $f2
+	charmap "<PERIOD>", $f2 ; same as "." in English
```
Comment (yenatch) [inline macros/charmap.asm]: `·` is a different character (https://en.wikipedia.org/wiki/Interpunct). `<DOT>` might be better.

Comment (roukaour) [reply macros/charmap.asm]: The tile looks identical to the period; it's used as a decimal point in PrintNum, and sometimes to abbreviate "`<ID>№`" (other times a period or slash is used). I'll rename it to "`<DOT>`".

Comment (yenatch) [reply macros/charmap.asm]: hmm, which one am i thinking of then. there's one that is a dot in the middle. or was that the ellipsis.

Comment (roukaour) [reply macros/charmap.asm]: You're thinking of `charmap "·", $74 ; middle dot, unused` in gfx/misc/font_extra.png.

**Comment by Guernouille** [comment]
I'm getting a lot of "already defined" errors from the NEXTU ENDU and UNION constants after this merge

**Comment by yenatch** [comment]
`UNION` was added in rgbasm 0.3.3. If you have an earlier version it will think it's a label and spit out that error.

**Comment by Guernouille** [comment]
Ooooooh. I'll install the new version then. Thank you!

**Pull request merged**